configuration-management.yaml

```diff
@@ -0,0 +1,302 @@
+                  - $ref: '#/components/schemas/DetailedError'
+        "401":
+          description: |
+            Unauthorized  
+            types: JwtAuthorization, LoginTokenExpired
+          content:
+            application/json:  
+              schema:
+                $ref: '#/components/schemas/GenericError'
+        "403":
+          description: |
+            Forbidden  
+            types: NotEnoughPrivileges
+          content:
+            application/json:  
+              schema:
+                $ref: '#/components/schemas/GenericError'
+        "422":
+          description: |
+            Unprocessable Entity  
+            types: LoginTokenSignatureInvalid
+          content:
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/GenericError'
+  /semester:
+    get:
+      tags:
+      - "Configuration Management"
+      summary: "Get the semester of the given date"
+      description: "Get which semester corresponds to the given date"
+      operationId: "getSemester"
+      parameters:
+      - name: "date"
+        in: "query"
+        description: "Date for which the semester needs to be fetched"
+        required: false
+        schema:
+          type: 'string'
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JsonSemester'
+              examples:
+                ConfigurationExample:
+                  $ref: '#/components/examples/JsonSemester'
+        "400":
+          description: | 
+            Bad Request  
+            types: QueryParameter
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DetailedError'
+  /validation:
+    get:
+      tags:
+      - "Configuration Management"
+      summary: "Get the regular expressions used for validation"
+      description: "Fetch the regular expressions used in the validaiton of various objects"
+      operationId: "getValidation"
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidationConfiguration'
+components:
+  securitySchemes:
+    uc4_token_login:
+      type: apiKey
+      in: cookie
+      name: login
+    uc4_token_login_header:
+      type: http
+      scheme: bearer
+  schemas:
+    Configuration:
+      type: "object"
+      properties:
+        fieldsOfStudy:
+          type: array
+          items:
+            type: string
+        countries:
+          type: array
+          items:
+            type: string
+        languages:
+          type: array
+          items:
+            type: string
+        courseTypes:
+          type: array
+          items:
+            type: string
+    ValidationConfiguration:
+      type: "object"
```

Comment:
Will this always be a flat object? Or does it make sense to change it to make address a field of user? I do not have any preference.

Reply:
I support using flat objects. While nesting the object makes the json slightly better looking, while resulting in atrocious looking code in our backend.